string.c

```diff
@@ -2873,15 +2873,29 @@ rb_str_append(VALUE str, VALUE str2)
     return rb_str_buf_append(str, str2);
 }
 
+#define MIN_PRE_ALLOC_SIZE 48
+
 VALUE
 rb_str_concat_literals(size_t num, const VALUE *strary)
 {
     VALUE str;
-    size_t i;
+    size_t i, s;
+    long len = 1;
+
+    if (UNLIKELY(!num)) return rb_str_new(0, 0);
+    if (UNLIKELY(num == 1)) return rb_str_resurrect(strary[0]);
```

Comment:
Thanks for your comment! :smile:

If `rb_str_concat_literals` is called from `concatstrings`(YARV insn) and it is executed as YARV insns generated by Ruby compiler, `num` seems not to be 0 nor 1.
But, if `rb_str_concat_literals` is called directly in ruby internal, `num` may be 0 or 1. 
I considered that situation.

But, `rb_str_concat_literals` is used only in `concatstrings` now, it may be better to reject them 💡